src/main/java/io/fabric8/maven/docker/model/ContainerDetails.java

```diff
@@ -121,6 +126,21 @@ public boolean isRunning() {
         return state.getBoolean(RUNNING);
     }
 
+    @Override
+    public boolean isHealthy() {
+        final JSONObject state = json.getJSONObject(STATE);
+        // always indicate healthy for docker hosts that do not support health checks.
+        return !state.has(HEALTH) || HEALTH_STATUS_HEALTHY.equals(state.getJSONObject(HEALTH).getString(STATUS));
+    }
+
+    @Override
+    public String healthcheckTests() {
+        if (!json.getJSONObject(CONFIG).has(HEALTHCHECK)) {
+            return null;
+        }
+        return json.getJSONObject(CONFIG).getJSONObject(HEALTHCHECK).getJSONArray(TEST).join(", ");
```

Comment:
Good catch, docker api might leave the test field undefined if the array is empty - however an empty test does not really make sense. I'll add a precondition.

We could ask `DockerAccess.getServerApiVersion` before each `getHealthyWaitChecker` and ensure a minimal api version (happens on other places too). I wasn't sure if it would be worth the overhead since we are backward-compatible anyway (by not waiting if there is no health status).